src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -79,35 +79,33 @@ public BitArray(byte[] bytes)
                 throw new ArgumentException(SR.Format(SR.Argument_ArrayTooLarge, BitsPerByte), nameof(bytes));
             }
 
-            m_array = new int[GetArrayLength(bytes.Length, BytesPerInt32)];
+            m_array = new int[GetInt32ArrayLengthFromByteLength(bytes.Length)];
             m_length = bytes.Length * BitsPerByte;
 
-            int i = 0;
-            int j = 0;
-            while (bytes.Length - j >= 4)
+            uint totalCount = (uint)bytes.Length / 4;
+
+            ReadOnlySpan<byte> byteSpan = bytes;
+            for (int i = 0; i < totalCount; i++)
             {
-                m_array[i++] = (bytes[j] & 0xff) |
-                              ((bytes[j + 1] & 0xff) << 8) |
-                              ((bytes[j + 2] & 0xff) << 16) |
-                              ((bytes[j + 3] & 0xff) << 24);
-                j += 4;
+                m_array[i] = MemoryMarshal.Read<int>(byteSpan);
```

Comment:
This will break Mono. It does not work on big-endian platforms. Use `BinaryPrimitives.ReadInt32LittleEndian` instead.
.

Reply:
Why did it work before when we were directly or'ing elements of the byte array?

Regardless of endianness, we had the following before, no?
**byte array:** `[a, b, c, d]`
**resulting int:** `[a | b << 8 | c << 16 | d << 24]` => `[d c b a]`

Why isn't the resulting int incorrect for big-endian? What am I missing here?

Don't we need to swap the ordering for big-endian?
**resulting int:** `[d | c << 8 | b << 16 | a << 24]` => `[a b c d]`

Reply:
> Never mind. The bytes in the byte array themselves are stored in reverse ordering when we are on a big-endian platforms. That's why we didn't need to reverse the or'ing order ourselves.

Right.

Reply:
Is there a way to add tests or CI legs that simulate big-endian platforms so that we can catch such mistakes?

Reply:
There is not one today

Reply:
@dotnet/dnceng, is it possible to add a big-endian CI leg, even if it is on demand only?

Reply:
@ahsonkhan This is not a problem for dnceng to solve. .NET Core or CoreFX does not run on any big-endian platform today directly. Bigendian systems are supported by Mono only.

Reply:
> This is not a problem for dnceng to solve.

Pardon my ignorance (and the ping). Thanks for clarifying, @jkotas. Good to know.